lib/postprocessing/image.js

```diff
@@ -87,23 +85,49 @@ function hasExtraInstructions(instructions) {
 }
 
 async function getImageMetadata(file) {
-    const start = new Date();
     try {
-        // [1x1+0+0] is a trick to avoid reading all pixels into memory, improving time for large images
-        const result = await gm(`${file}[1x1+0+0]`).write("json:-");
-        const json = JSON.parse(result.toString());
+        // suffix attributes with `#` to disable human-readable parsing if exiftool not used with `-n` flag
+        const attributes = [
+            "-Orientation",
+            "-FileType",
+            "-ImageHeight",
+            "-ImageWidth",
+            "-JPEGQualityEstimate"
+        ];
+        const attributesToExtract = attributes.join(" ");
+        const command = `exiftool -n -json ${attributesToExtract} ${file}`;
+        console.log(`Metadata (exiftool) command is: ${command}`);
+
+        const { stdout, stderr } = await exec(command);
+
+        if(stderr){
+            throw new GenericError(`Reading metadata from intermediate rendition failed: ${stderr}`);
+        }
 
-        // newer imagemagick returns an array (for multi-page formats), older just one object (6.8.9-9 at least)
-        if (Array.isArray(json)) {
-            return json[0].image;
-        } else {
-            return json.image;
+        let metadata = JSON.parse(stdout);
+        if (Array.isArray(metadata)) {
+            metadata = metadata[0];
         }
+
+        if(!metadata){
+            throw new GenericError("No metadata read from intermediate rendition");
+        }
+
+        // normalize to have nicer names (and same names as for imagemagick's metadata reader)
+        const filteredMetadata = {};
+        filteredMetadata.format = metadata.FileType;
+        filteredMetadata.pageGeometry = {};
+        filteredMetadata.pageGeometry.height = metadata.ImageHeight;
+        filteredMetadata.pageGeometry.width = metadata.ImageWidth;
+        filteredMetadata.orientation = metadata.Orientation;
+        filteredMetadata.quality = metadata.JPEGQualityEstimate;
```

Comment:
I guess we want to change all usages to use the exiftool json output. The one from imagemagick is entirely undocumented, so future maintainers of this code (including us) would have a hard time figuring out what this intermediate object structure (`filteredMetadata`) should be.

Reply:
I would prefer not to. I change `.pageGeometry.height` to `.imageHeight` and `.pageGeometry.width` to `.imageWidth`. But I'd appreciate we don't change the naming depending on every tool we use - hence that very small and basic abstraction.
Also, the object structure is kept flat on purpose.

Reply:
Please use the exiftool format. That is documented or at least clear. The new one you added is custom and has no documentation. So when new metadata fields are added folks have to invent new stuff and will get confused. Just extracting all metadata and using the exiftool format is good. We do expect to stick with exiftool now :-)

Reply:
Just extracting all metadata and using the exiftool format is good. <= No, for reasons explained in https://github.com/adobe/asset-compute-sdk/pull/78#discussion_r520676816, https://github.com/adobe/asset-compute-sdk/pull/78#discussion_r520678938 and https://github.com/adobe/asset-compute-sdk/pull/78#discussion_r520883756. 

Fine for naming. It's a bit a leaky abstraction, but I don't feel like fighting naming here.